core/src/oracle_config.rs

```diff
@@ -41,15 +41,26 @@ pub struct OracleConfig {
 
 pub struct OracleSecrets {
     pub node_api_key: String,
+    pub wallet_password: Option<String>,
 }
 
 impl OracleSecrets {
     pub fn load() -> Self {
-        std::env::var("ORACLE_NODE_API_KEY")
-            .map(|node_api_key| Self { node_api_key })
+        let api_key = std::env::var("ORACLE_NODE_API_KEY").unwrap_or_else(|_| {
+            panic!("ORACLE_NODE_API_KEY environment variable for node API key is not set")
+        });
+
+        let wallet_pass = std::env::var("ORACLE_NODE_WALLET_PASSWORD")
+            .map(Some)
```

Comment:
Can use `Result.ok` to map `Result` into `Option`:

```rust
let wallet_pass = std::env::var("ORACLE_NODE_WALLET_PASSWORD").ok();
if wallet_pass.is_none() {
    warn!("ORACLE_NODE_WALLET_PASSWORD environment variable for automatic unlock of node wallet is not set");
}
```

Reply:
Adjusted in recent commit.